### PR TITLE
podman: support --userns=ns|container

### PR DESCRIPTION
**Pull request by giuseppe**
allow to join the user namespace of another container.

Closes: https://github.com/containers/libpod/issues/3629

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by mheon** [comment]
This LGTM

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, tests are still not hip

**Comment by mheon** [comment]
Play kube failures probably legitimate - the way we set up the createconfig there is different from Podman's normal behavior, which is probably tripping our userns detection

**Comment by giuseppe** [comment]
> Play kube failures probably legitimate - the way we set up the createconfig there is different from Podman's normal behavior, which is probably tripping our userns detection

tests are finally green

File: cmd/podman/shared/create.go [hunk 1]
```diff
@@ -282,13 +282,26 @@ func configurePod(c *GenericCLIResults, runtime *libpod.Runtime, namespaces map[
 	if err != nil {
 		return namespaces, err
 	}
+	hasUserns := false
+	if podInfraID != "" {
+		podCtr, err := runtime.LookupContainer(podInfraID)
```
Comment (mheon) [inline cmd/podman/shared/create.go]: Don't need a lookup here - it's guaranteed to be full ID, so we can do a straight `runtime.Container()`

Comment (giuseppe) [reply cmd/podman/shared/create.go]: thanks.  Pushed a new version

**Pull request merged**